tensorflow_datasets/image/colorectal_histology.py

```diff
@@ -0,0 +1,169 @@
+_LARGE_SUBDIR = "Kather_texture_2016_larger_images_10"
+
+_CLASS_NAMES = (
+    "tumor",
+    "stroma",
+    "complex",
+    "lympho",
+    "debris",
+    "mucosa",
+    "adipose",
+    "empty",
+)
+_TILES_SIZE  = 150
+_LARGE_SIZE = 5000
+_SEED = 37  # used to determine train/val/test splits
+
+_CITATION = """\
+@article{kather2016multi,
+  title={Multi-class texture analysis in colorectal cancer histology},
+  author={Kather, Jakob Nikolas and Weis, Cleo-Aron and Bianconi, Francesco and Melchers, Susanne M and Schad, Lothar R and Gaiser, Timo and Marx, Alexander and Z{\"o}llner, Frank Gerrit},
+  journal={Scientific reports},
+  volume={6},
+  pages={27988},
+  year={2016},
+  publisher={Nature Publishing Group}
+}
+"""
+
+def _class_subdir(class_index, class_name):
+  return "%02d_%s" % (class_index + 1, class_name.upper())
+
+
+def _load_tif(path):
+  try:
+    from PIL import Image
+  except ImportError:
+    raise ImportError(
+        "PIL required for colorectal histology dataset encoding")
+
+  with tf.io.gfile.GFile(path, "rb") as fp:
+    image = Image.open(fp)
+  return np.array(image)
+
+
+class ColorectalHistology(tfds.core.GeneratorBasedBuilder):
+  """Biological 8-class classification problem."""
+  URL = _URL
+  VERSION = tfds.core.Version("0.0.1")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=(
+          "Classification of textures in colorectal cancer histology. "
+          "Each example is a 150 x 150 x 3 RGB image of one of 8 classes."),
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=(_TILES_SIZE,)*2 + (3,)),
+            "label": tfds.features.ClassLabel(
+                names=_CLASS_NAMES),
+            "filename": tfds.features.Text(),
+        }),
+        urls=[_URL],
+        citation=_CITATION,
+        supervised_keys=("image", "label"),
+    )
+
+  def _split_generators(self, dl_manager):
+    folder = dl_manager.download_and_extract(_TILES_DL_URL)
+    # no official train/validation/test split, so we use 80% / 10% / 10%
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=8,
+            gen_kwargs=dict(
+                start_frac=0,
+                end_frac=0.8,
+                root_dir=folder
+            )
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=1,
+            gen_kwargs=dict(
+                start_frac=0.8,
+                end_frac=0.9,
+                root_dir=folder
+            )
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=1,
+            gen_kwargs=dict(
+                start_frac=0.9,
+                end_frac=1,
+                root_dir=folder
+            )
+        ),
+    ]
+
+  def _generate_examples(self, root_dir, start_frac, end_frac):
+    random.seed(_SEED)
```

Comment:
This would set the random seed globally and may overwrite the any user defined seed.

Here you can remove it as tfds will automatically shuffle the examples for you.

In the general case, you should be using a local generator instead.
```
r = random.Random(seed=123)
r.shuffle(my_list)
```